bindgen/TypeTranslator.h

```diff
@@ -20,14 +14,38 @@ class TypeTranslator {
      * structs, unions, ...
      * @return the type translated
      */
-    std::string Translate(const clang::QualType &tpe,
-                          const std::string * = nullptr);
-    std::string TranslateFunctionPointer(const clang::QualType &qtpe,
-                                         const std::string *avoid);
-    std::string TranslatePointer(const clang::QualType &pointee,
+    Type *translate(const clang::QualType &tpe, const std::string * = nullptr);
+
+    void addAlias(std::string cName, Type *type);
+
+    std::string getTypeFromTypeMap(std::string cType);
+
+  private:
+    clang::ASTContext *ctx;
+    IR &ir;
+
+    /**
+     * Primitive types
+     */
+    std::map<std::string, std::string> typeMap;
+
+    /**
+     * Maps C struct, union or enum name to Type alias
+     */
+    std::map<std::string, Type *> aliasesMap;
```

Comment:
I quite liked the unification in the original `typeMap`. Would it make sense to change it to be defined as:
```c++
std::map<std::string, Type *> typeMap
```
populated with
```c++
     typeMap["void"] = new PrimitiveType("Unit");
     typeMap["bool"] = new PrimitiveType("native.CBool");
     typeMap["_Bool"] = new PrimitiveType("native.CBool");
     // ...
```
I assume the problem is with the deallocation?

Reply:
Yes, deallocation is the problem (well, these instances are not deallocated now, but changing type of the map will make fixing it harder)

Reply:
I'll create an issue for deallocation

Reply:
Or simply do it now, because I do not want the code that does not deallocate objects